lib/sproxyd.js

```diff
@@ -187,9 +187,21 @@ class SproxydClient {
                 log.debug('stored to sproxyd', { newKey });
                 return callback(null, newKey);
             });
+            request.on('finish',
+                () => log.debug('finished sending PUT chunks to sproxyd', {
+                    component: 'sproxydclient',
+                    method: '_handleRequest',
+                    contentLength: size,
+                }));
             stream.pipe(request);
-            request.on('end', () => request.end);
-            log.debug('finished sending PUT chunks to sproxyd');
+            stream.on('error', err => {
+                log.error('error from readable stream', {
+                    error: err,
+                    method: '_handleRequest',
+                    component: 'sproxydclient',
+                });
+                request.end();
```

Comment:
does it have the effect on the writable stream of closing the underlying tcp/ip connection to the remote host? if not, the remote host will wait until timeout for the remaining data to reach 'content-length'. (not good for stuffs like http keep-alive)